app/assets/javascripts/index/directions.js

```diff
@@ -51,7 +43,7 @@ OSM.Directions = function (map) {
     select.append("<option value='" + i + "'>" + I18n.t("javascripts.directions.engines." + engine.id) + "</option>");
   });
 
-  function Endpoint(input, iconUrl, dragCallback, geocodeCallback) {
+  function Endpoint(input, iconUrl) {
```

Comment:
You're undoing #4902. I was going to move endpoint code into a separate module because it's difficult to test this one.

Reply:
I apologize for the undoing, I thought it's just simplifying .. I returned using callback for calling getRoute, so, Endpoint can be pulled out now to separate module.

Thanks,
Nenad.